lib/oplog/oplog_v2_converter.go

```diff
@@ -186,8 +198,16 @@ func getChangedFieldsFromOplogV2Update(op *oplogEntry) []string {
 	}
 
 	if config.OplogV2ExtractSubfieldChanges() {
+		var diffMap map[string]interface{}
+		err := bson.Unmarshal(diffRaw, &diffMap)
+		if err != nil {
+			metricUnprocessableChangedFields.Inc()
+			log.Log.Errorw("Oplog data for non-replacement v2 update had a diff that was not a map",
+				"op", op)
+			return []string{}
+		}
 		return getChangedFieldsFromOplogV2UpdateDeep(diffMap, "")
```

Comment:
Why do the unmarshalling here versus updating the deep version of the function to also use the raw bson?

Reply:
Since it's going deeper, there's a little less to gain, but there's a lot more complexity.  I also don't know if we normally use the deep mode or not -- it seems it's set by an env var.  I'll try to find out more to see if this would be a worthwhile addition.

Reply:
I checked and the current tulip charts for OTR don't even pass in the env to enable the deep change fields (OPLOG_V2_EXTRACT_SUBFIELD_CHANGES), and it defaults to false, so we're currently not using this code.  It would be more complete to include it, but I think at least for now it's ok to skip it and prove the value of this update first.

Reply:
I did add an error that will bubble up to the level where the original unmarshal was